### PR TITLE
Case 21474: relative paths in FST material parsing

### PR DESCRIPTION
**Pull request by thoys**
https://highfidelity.manuscript.com/f/cases/21474/Embedded-fst-Materials-should-support-relative-paths

### CODE REVIEW

File: libraries/model-baker/src/model-baker/Baker.cpp [hunk 1]
```diff
@@ -155,7 +155,8 @@ namespace baker {
             const auto jointIndices = jointInfoOut.getN<PrepareJointsTask::Output>(2);
 
             // Parse material mapping
-            const auto materialMapping = model.addJob<ParseMaterialMappingTask>("ParseMaterialMapping", mapping);
+            const auto parseMaterialMappingInputs = ParseMaterialMappingTask::Input(url, mapping).asVarying();
```
Comment (SamGondelman) [inline libraries/model-baker/src/model-baker/Baker.cpp]: Just kidding, this is the wrong url

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17715/HighFidelity-Beta-PR15049-de3c5e0.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17715/HighFidelity-Beta-Interface-PR15049-de3c5e0.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17715/HighFidelity-Beta-PR15049-de3c5e0.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17715/HighFidelity-Quest-Beta-PR15049-de3c5e0.apk)

File: libraries/model-networking/src/model-networking/ModelCache.cpp [hunk 2]
```diff
@@ -78,7 +80,7 @@ namespace std {
     struct hash<GeometryExtra> {
         size_t operator()(const GeometryExtra& geometryExtra) const {
             size_t result = 0;
-            hash_combine(result, geometryExtra.mapping, geometryExtra.textureBaseUrl, geometryExtra.combineParts);
+            hash_combine(result, geometryExtra.mapping.second, geometryExtra.textureBaseUrl, geometryExtra.combineParts);
```
Comment (SamGondelman) [inline libraries/model-networking/src/model-networking/ModelCache.cpp]: I wonder if we need to do this:
```suggestion
            hash_combine(result, geometryExtra.mapping.first, geometryExtra.mapping.second, geometryExtra.textureBaseUrl, geometryExtra.combineParts);
```
and you might need to add an std hash for QUrl

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17745/HighFidelity-Beta-PR15049-6aede02.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17745/HighFidelity-Quest-Beta-PR15049-6aede02.apk)

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17745/HighFidelity-Beta-PR15049-6aede02.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17745/HighFidelity-Beta-Interface-PR15049-6aede02.dmg

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17750/HighFidelity-Beta-PR15049-ff8cb27.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17750/HighFidelity-Quest-Beta-PR15049-ff8cb27.apk)

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17745/HighFidelity-Beta-PR15049-6aede02.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17745/HighFidelity-Beta-Interface-PR15049-6aede02.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17750/HighFidelity-Beta-PR15049-ff8cb27.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17750/HighFidelity-Beta-Interface-PR15049-ff8cb27.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17750/HighFidelity-Beta-PR15049-ff8cb27.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17750/HighFidelity-Beta-Interface-PR15049-ff8cb27.exe

**Pull request merged**